src/context.cc

```diff
@@ -59,6 +59,7 @@ Context::Context(const Context *parent)
 	}
 	else {
 		this->ctx_stack = new Stack;
+		this->document_path = std::make_shared<std::string>();
```

Comment:
I'm hesitating a bit due to this creating an empty `shared_ptr`. It does not seem to be dereferenced anywhere as the tests are green. Do you see a way to make it more obvious that the value is always set, e.g. by passing it along in the constructor? It still needs a setter for overwriting for the `import` modules though.

Reply:
It makes a shared_ptr to an empty string, if I'm not mistaken. Maybe I can make that more obvious like this: ```std::make_shared<std::string>("")```? Let's see if it can be set in the initializer list.
W.r.t. the setter, how do you mean? ```Context::setDocumentPath``` is there, and in ```FileContext::initializeModule``` the shared_ptr reassignment works its magic.
I'll do some more testing.

Reply:
Ahh, right, looks like it does create an empty string in that case (https://godbolt.org/z/6jUVWf). I guess that makes it really identical to the old version minus the copy overhead.